bin/check-dns.rb

```diff
@@ -132,6 +133,19 @@ def check_against_regex(entries, regex)
     critical "Resolved #{config[:domain]} #{config[:type]} did not match #{regex}"
   end
 
+  def check_ips(entries)
+    ips = []
+    entries.answer.rrsets(config[:type]).each do |ip|
+      ips << ip.to_s.split("\t").last.tr("\n", '')
+    end
+    result = (IPAddr.new config[:result]).to_s
+    if ips.any? { |ip| ip.casecmp(result.downcase).zero? }
```

Comment:
`.casecmp` and `.downcase` is odd to include here. You should be able to just compare an `IPAddr` to these IPs, assuming you don't call `.to_s` on `result`:

    if ips.any? { |ip| result == ip }